enhancements/baremetal-IPI/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,185 @@
+  - "@abhinavdahiya"
+  - "@dhellmann"
+
+approvers:
+  - "@abhinavdahiya"
+  - "@enxebre"
+  - "@deads2k"
+
+creation-date: 2019-11-14
+last-updated: 2019-11-15
+status: implemented
+see-also:
+  - 
+replaces:
+  - 
+superseded-by:
+  - 
+---
+
+# Config required for Baremetal IPI deployments
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions
+
+1. There are no issues reading this new CR within operator scope from the
+Machine API Operator.
+
+2. Naming the new CR.
+
+3. Since this enhancement did not make it into 4.3 and the BareMetal IPI
+deployments are in Tech Preview in 4.2, the upgrade strategy should include
+upgrades from 4.2 to 4.3 to 4.4. (Assuming this enhancement makes it into 4.4.)
+
+## Summary
+
+The configuration required to bring up a Baremetal IPI (metal3) cluster needs
+to be availabe in a Config Resource. We had submitted an earlier proposal [1]
+to augment the BareMetalInfrastructureStatus CR with these config items. Based
+on feedback we recieved we are revising our proposal to create a new CR for
+these configuration items.
+
+This new CR needs to be accessed by just the Installer and the Machine API
+Operator (API) and hence does not need to be created with global scope.
+
+## Motivation
+
+The Baremetal IPI deployments are different from the other platform types currently
+being supported by OpenShift in that there is no underlying cloud controller
+that exists until provisioning of baremetal servers takes place first.
+
+The "metal3" pod deployed by the machine API operator (MAO) contains the BaremMetal
+Operator (BMO) and a provisioning service (Ironic) that are together responsible for
+PXE booting baremetal servers and enrolling them as BareMetal Hosts.
+
+This enhancement request proposes to add a new CR for configs required only for
+the provisioning service to PXE boot a baremetal server and make it available as a
+BareMetal host.The configurations in this baremetal provisioning CR, are seperate
+from the configuration in the BareMetalHost CR. This new CR will allow for setting
+some sane defaults for these configurations with the ability to overrride them if 
+required. After the baremetal server is provisioned, the configurations in the
+BareMetalHost CR contain information to connect to the controller and also to
+fulfill a Machine.
+
+### Goals
+
+The goal of this enhancement request is to provide details about the configuration
+being added to a new CR used only in "metal3" deployments. This new CR would be
+within the operator scope and will only contain contain configuration required
+by the provisioning service to boot baremetal servers.
+
+### Non-Goals
+
+All config values except the RHCOSImageUrl are not expected to change after a
```

Comment:
Is there any reason we would want to prevent them from changing? Or do we just think they won't?

Reply:
This is a good point, while we probably don't want to support changing the provisioning network subnet completely, it probably is reasonable to e.g extend the DHCP pool size.

That said, it's probably true to say we've not explored these kind of updates yet, and @sadasu has mentioned that below so I'd be fine to say for this first-pass we just expect to write the data at install time, and update support for the individual elements can be tested/documented as a follow-up task?